src/main/java/org/gridsuite/modification/server/dto/NetworkModificationResult.java

```diff
@@ -47,10 +49,18 @@ public ApplicationStatus max(ApplicationStatus other) {
 
     @Schema(description = "Network modification impacts")
     @Builder.Default
-    private List<SimpleElementImpact> networkImpacts = List.of();
+    private List<AbstractBaseImpact> networkImpacts = List.of();
 
     public Set<String> getImpactedSubstationsIds() {
-        return networkImpacts.stream().flatMap(impact -> impact.getSubstationIds().stream()).collect(Collectors.toCollection(TreeSet::new));
+        Set<String> ids = new TreeSet<>();
+        networkImpacts.stream().forEach(impact -> {
+            if (impact instanceof SimpleElementImpact simpleImpact) {
+                ids.addAll(simpleImpact.getSubstationIds());
+            } else if (impact instanceof CollectionElementImpact) {
+                // TODO: Do nothing for now
+            }
```

Comment:
Why don't you do a filter + map + collect?

Reply:
It's the previous code, I will